src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1312,16 +1312,20 @@ private function dumpValue($value, $interpolate = true)
                 }
 
                 if (is_array($factory)) {
+                    $factoryMethod = substr($this->dumpValue($factory[1]), 1, -1);
+
                     if (is_string($factory[0])) {
-                        return sprintf('\\%s::%s(%s)', $factory[0], $factory[1], implode(', ', $arguments));
+                        $factoryClass = substr(str_replace('\\\\', '\\', $this->dumpValue($factory[0])), 1, -1);
```

Comment:
Hm, would it make more sense to completely move the logic to the compiler pass? And is it actually possible to move it in a BC way?